assets/prototype/domain/shared/predicates/tickets/filters/nextOnSaleOrPendingOnly.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * External dependencies
+ */
+import { allPass, filter, head } from 'ramda';
+
+/**
+ * Internal dependencies
+ */
+import { isOnSale, isPending } from '../index';
+import sortTicketEntitiesList from './sortTicketEntitiesList';
+
+const nextOnSaleOrPendingOnly = (ticketEntities: any[]) => {
+	const selectedPredicates = [isOnSale, isPending];
+	const onSaleAndPending = filter(allPass(selectedPredicates))(ticketEntities);
+	const sortedItems = sortTicketEntitiesList(onSaleAndPending);
+
+	return [head(sortedItems)];
```

Comment:
I am not very happy with the readability here.
On the first iteration, I've just mimicked the old implementation.
On the next iteration, when coupled with UI, this needs to be improved/ simplified.